static/scripts/rewards/web3/erc20-permit.ts

```diff
@@ -100,12 +100,17 @@ async function transferFromPermit(permit2Contract: Contract, app: AppState) {
 async function waitForTransaction(tx: TransactionResponse) {
   try {
     const receipt = await tx.wait();
+    viewClaimButton.onclick = () => {
+      window.open(`https://${tx.chainId === 1 ? "etherscan" : "gnosisscan"}.io/tx/${receipt.transactionHash}`, "_blank");
```

Comment:
Seems like a bad approach. Why would you only support one or the other if we are supposed to support every EVM chain with a permit2 deployment?

Reply:
Good point, the most robust solution I could come up with is to use [Blockscan](https://blockscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3). Check the link, it's the Permit2 address and on the left hand side there is a dropdown of the chains that are supported (26).

[Blockscan TX](https://blockscan.com/tx/0x1fde287d41edd78b291c3e9f3f4ac7a53a1f10a14c9f61d25913ac2e95814f8b)

Reply:
Manually collect seems fine. I can't find the table of networks from my phone but check the uniswap docs.

Reply:
List of supported chains through blockscan

[Etherscan (Ethereum)](https://etherscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Arbiscan (Arbitrum One)](https://arbiscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Polygonscan (Polygon)](https://polygonscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[BscScan (BNB Chain)](https://bscscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Optimism (Optimism)](https://optimistic.etherscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[BaseScan (Base)](https://basescan.org/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[BttcScan (BTTC)](https://bttcscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[CeloScan (Celo)](https://celoscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[FTMScan (Fantom)](https://ftmscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Gnossiscan (Gnosis)](https://gnosisscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Polygon zkEVM (Polygon zkEVM)](https://zkevm.polygonscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[KromaScan (Kroma)](https://kromascan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[LineaScan (Linea)](https://lineascan.build/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Moonbeam (Moonbeam)](https://moonbeam.moonscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Moonriver (Moonriver)](https://moonriver.moonscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Arbitrum Nova (Arbitrum Nova)](https://nova.arbiscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[ScrollScan (Scroll)](https://scrollscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[WemixScan (Wemix)](https://wemixscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[SnowScan (Avax C-Chain)](https://snowscan.xyz/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[zkSync Era (zkSync Era)](https://era.zksync.network/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[opBNB (opBNB)](https://opbnb.bscscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[FraxScan (Fraxtal)](https://fraxscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[BlastScan (Blast)](https://blastscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Cronos Explorer (Cronos)](https://cronoscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Mantle Explorer (Mantle)](https://mantlescan.xyz/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

[Taiko Explorer (Taiko)](https://taikoscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3)

Reply:
Best to open an issue in `rpc-handler` to update the network explorers and then use that as a single source across projects then

Reply:
I just quickly looked up where it has activity. 

https://blockscan.com/address/0x000000000022d473030f116ddee9f6b43ac78ba3#portfolios

![blockscan com_address_0x000000000022d473030f116ddee9f6b43ac78ba3](https://github.com/ubiquity/pay.ubq.fi/assets/4975670/238a6c5f-87e6-40ab-991b-7d3d64e2bbff)

We should manually test that it works on the main chains, such as: Ethereum mainnet, Optimism, Arbitrum, Polygon, Gnosis, Base,